internal/lvmd/command/lvm_lv_attr.go

```diff
@@ -0,0 +1,283 @@
+	AllocationPolicyInherited        AllocationPolicy = 'i'
+	AllocationPolicyInheritedLocked  AllocationPolicy = 'I'
+	AllocationPolicyCling            AllocationPolicy = 'l'
+	AllocationPolicyClingLocked      AllocationPolicy = 'L'
+	AllocationPolicyNormal           AllocationPolicy = 'n'
+	AllocationPolicyNormalLocked     AllocationPolicy = 'N'
+	AllocationPolicyNone                              = '-'
+)
+
+type Minor rune
+
+const (
+	MinorTrue  Minor = 'm'
+	MinorFalse Minor = '-'
+)
+
+type State rune
+
+const (
+	StateActive                                State = 'a'
+	StateSuspended                             State = 's'
+	StateInvalidSnapshot                       State = 'I'
+	StateSuspendedSnapshot                     State = 'S'
+	StateSnapshotMergeFailed                   State = 'm'
+	StateSuspendedSnapshotMergeFailed          State = 'M'
+	StateMappedDevicePresentWithoutTables      State = 'd'
+	StateMappedDevicePresentWithInactiveTables State = 'i'
+	StateNone                                  State = '-'
+	StateHistorical                            State = 'h'
+	StateThinPoolCheckNeeded                   State = 'c'
+	StateSuspendedThinPoolCheckNeeded          State = 'C'
+	StateUnknown                               State = 'X'
+)
+
+type Open rune
+
+const (
+	OpenTrue    Open = 'o'
+	OpenFalse   Open = '-'
+	OpenUnknown Open = 'X'
+)
+
+type OpenTarget rune
+
+const (
+	OpenTargetMirror   = 'm'
+	OpenTargetRaid     = 'r'
+	OpenTargetSnapshot = 's'
+	OpenTargetThin     = 't'
+	OpenTargetUnknown  = 'u'
+	OpenTargetVirtual  = 'v'
+	OpenTargetNone     = '-'
+)
+
+type Zero rune
+
+const (
+	ZeroTrue  Zero = 'z'
+	ZeroFalse Zero = '-'
+)
+
+type VolumeHealth rune
+
+const (
+	VolumeHealthPartialActivation        = 'p'
+	VolumeHealthUnknown                  = 'X'
+	VolumeHealthOK                       = '-'
+	VolumeHealthRAIDRefreshNeeded        = 'r'
+	VolumeHealthRAIDMismatchesExist      = 'm'
+	VolumeHealthRAIDWriteMostly          = 'w'
+	VolumeHealthRAIDReshaping            = 's'
+	VolumeHealthRAIDReshapeRemoved       = 'R'
+	VolumeHealthThinFailed               = 'F'
+	VolumeHealthThinPoolOutOfDataSpace   = 'D'
+	VolumeHealthThinPoolMetadataReadOnly = 'M'
+	VolumeHealthWriteCacheError          = 'E'
+)
+
+type SkipActivation rune
+
+const (
+	SkipActivationTrue  SkipActivation = 'k'
+	SkipActivationFalse SkipActivation = '-'
+)
+
+// LVAttr has mapped lv_attr information, see https://linux.die.net/man/8/lvs
+// It is a complete parsing of the entire attribute byte flags that is attached to each LV.
+// This is useful when attaching logic to the state of an LV as the state of an LV can be determined
+// from the Attributes, e.g. for determining whether an LV is considered a Thin-Pool or not.
+type LVAttr struct {
+	VolumeType
+	Permissions
+	AllocationPolicy
+	Minor
+	State
+	Open
+	OpenTarget
+	Zero
+	VolumeHealth
+	SkipActivation
+}
+
+const lvAttrLength = 10
+
+func ParsedLVAttr(raw string) (LVAttr, error) {
```

Comment:
LVAttr is a large-size structure, why don't you use a pointer?

Reply:
is a struct with 10 runes (mostly 10*int32 that defaults to 0 value) really a large size struct? Especially because the zero value is not taking any space?

I can adjust to pointer without issues, but I didnt think it was necessary

Reply:
Even if they are zero values, the size is 40.
https://go.dev/play/p/FXmATPqrCYT

Anyway, we have no rule to return a struct, leave it as is now.

Reply:
I think its a valid concern and its easy to adjust, let me change to a pointer